controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1611,6 +1616,88 @@ bool master_thread::handle_cmdu_1905_topology_notification(const std::string &sr
     return true;
 }
 
+bool master_thread::handle_cmdu_1905_topology_response(const std::string &src_mac,
+                                                       ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received TOPOLOGY_RESPONSE_MESSAGE, mid=" << std::hex << int(mid);
+
+    auto tlvDeviceInformation = cmdu_rx.getClass<ieee1905_1::tlvDeviceInformation>();
+    if (!tlvDeviceInformation) {
+        LOG(ERROR) << "ieee1905_1::tlvDeviceInformation not found";
+        return false;
+    }
+
+    const auto &al_mac = tlvDeviceInformation->mac();
+
+    // Get all Agent known fronthaul radios if exist, and compare it to what the Agent reports.
+    // If a radio exist on the database but not the tlvDeviceInformation, this radio node needs
+    // to be removed.
+    // This shall work in the opposite way as well: If it doesn't exist on the database, but
+    // reported in the TLV, then we need to add the node to the database. Of course this will remain
+    // as "TODO" for future task. Meanwhile parse properly the TLV, leaving the unused parts in
+    // comment for future implementation.
+    auto fronthaul_radios_on_db = database.get_node_children(
```

Comment:
That's not just fronthaul radios; I think it's really AP radios that you mean (VAPs on that radio can be fronthaul and/or backhaul VAPs; the point it's that it's not backhaul STA).

Reply:
No, this tlv refers only to the radio interface and not vaps.
The vaps are handled in AP operational BSS TLV.

Reply:
Yes, that's what I mean: there is no such thing as a "fronthaul radio". There are fronthaul and/or backhaul VAPs on a radio, but every radio can be used as fronthaul or backhaul AP.

The only limitation you can have on a radio is that it is not going to be used as an AP.

Since this term "fronthaul radio" is introduced by this PR, I think it's rather important to *not* introduce it if it is not correct (or potentially confusing).

Reply:
Actually, the term "fronthaul radio" *is* used all over the place in the test plan, though it is not defined anywhere. But therefore it's OK to use it here as well.